lib/util/Components.js

```diff
@@ -253,34 +255,33 @@ function componentRule(rule, context) {
     },
 
     /**
-     * Check if createElement is destructured from React import
+     * Check if variable is destructured from pragma import
      *
-     * @returns {Boolean} True if createElement is destructured from React
+     * @param {variable} String The variable name to check
+     * @returns {Boolean} True if createElement is destructured from the pragma
      */
-    hasDestructuredReactCreateElement: function() {
+    isDestructuredFromPragmaImport: function(variable) {
       const variables = variableUtil.variablesInScope(context);
-      const variable = variableUtil.getVariable(variables, 'createElement');
-      if (variable) {
-        const map = variable.scope.set;
-        if (map.has('React')) {
-          return true;
-        }
+      const variableInScope = variableUtil.getVariable(variables, variable);
+      if (variableInScope) {
+        const map = variableInScope.scope.set;
+        return map.has(pragma);
       }
       return false;
     },
 
     /**
-     * Checks to see if node is called within React.createElement
+     * Checks to see if node is called within createElement from pragma
      *
      * @param {ASTNode} node The AST node being checked.
-     * @returns {Boolean} True if React.createElement called
+     * @returns {Boolean} True if createElement called from pragma
      */
-    isReactCreateElement: function(node) {
-      const calledOnReact = (
+    isPragmaCreateElement: function(node) {
```

Comment:
we could even simplify this name to `isCreateElement`?

Reply:
Sounds good